java/http/java-http-consumer/src/main/java/HttpConsumer.java

```diff
@@ -166,7 +167,9 @@ public void poll() {
 
                 ArrayNode records = (ArrayNode) MAPPER.readTree(response.body());
                 log.info("... got {} records", records.size());
-
+                if (records.size() > 0) {
+                    this.commitMessages();
+                }
```

Comment:
In general you should commit after the messages processing, where in our case processing means just printing in the log.

Reply:
@ppatierno Does this mean just moving the `commitMessages()` call below the for loop [1] in the `poll` method? 
```  for (JsonNode record : records) {
                    log.info("Record {}", record);
                    this.messageReceived++;
                } ```

Reply:
related to the specific comment yes, but if you read the main comment I left on the review, it's useless.